packages/cadl-ranch-specs/http/server/version/query/main.tsp

```diff
@@ -0,0 +1,33 @@
+import "@typespec/http";
+import "@typespec/versioning";
+import "@azure-tools/cadl-ranch-expect";
+
+using TypeSpec.Http;
+using TypeSpec.Versioning;
+
+@doc("Illustrates server with multiple versions and the version is in query.")
+@scenarioService("/server/version/query", Options)
+@server(
+  "{endpoint}",
+  "Testserver endpoint",
+  {
+    @doc("Need to be set as 'http://localhost:3000' in client.")
+    endpoint: url,
+  }
+)
+namespace Server.Version.Query;
+
+enum Versions {
+  v1_1: "v1.1",
+  v1_2: "v1.2",
+}
+
+model Options {
+  versioned: Versions;
+}
+
+@scenario
+@scenarioDoc("An simple operation in a parameterized server.")
+@route("/myOp")
+@head
+op myOp(): OkResponse;
```

Comment:
how does it resolve the version here? there is no reference to it

Reply:
Please consider a better name. `myOp` shows on dashboard, and it be meaningless.

Also describe what you want the tester to do in `scenarioDoc`. What's the input (query/body) you are expecting?

Reply:
The generated url at sdk side should contains the version. See my updated description.

Reply:
If I understand correctly, this case doesn't care about how service evolves. I just want to make sure the api version I set would generate correct url. At SDK side, my expectation is when I initiate the client with v1.1, then the url the client would send request is  `/v1.1/server/version/path/valid`

Reply:
This is about the query spec, the path one looks good and does what I expect. The query one says it will send this url
```
Expected path with api version parameter: /server/version/query/valid?api-version=v1.1
```
but there is 0 reference to a parameter called `api-version` in the spec unless I'm missing it.

Reply:
`v1.1` is what I expect user to pass. For e.g., 
```C#
var response = await new QueryClient(host, new QueryClientOptions(QueryClientOptions.ServiceVersion.V1_1)).ValidAsync();
```
When user creates a class, he should indicate that the api version is `v1.1`. The fully generated client and corresponding test is in https://github.com/Azure/autorest.csharp/pull/3514 under folder [test/CadlRanchProjects/server/version/query/src/Generated/](https://github.com/Azure/autorest.csharp/pull/3514/files#diff-5541013fd5e3a4a736ac6e77f52fe25066f59ff294b899a7ca2529615934aa11)

Reply:
OK, seems you are saying this is not a valid case?

Our current behavior is if there is no api-version parameter in the operation, we will generate one, otherwise SDK cannot work.

Reply:
Depends on what you mean by valid case. That spec is a valid representation of some http service but it has no version being used.

That part "the SDK cannot work" is what needs to be worked on, not adding magic parameters that will prevent non azure SDK from working in the future and/or conflict with someone that end up depending on it implicitly.

Reply:
I edited my comment with the link to [AnmalyDetector](https://github.com/Azure/azure-rest-api-specs/blob/4854f07c1ff8b67930ad5bc6a3042ad73b241946/specification/cognitiveservices/AnomalyDetector/multivariate/routes.tsp).

Reply:
@m-nash  Currently for an operation without api-version parameter like 
```
op valid(): NoContentResponse;
```
We will add the api-version parameter to it. But Tim mentioned it will prevent non azure SDK from working in the future. Do we have plan to support non azure SDK?

Reply:
Like in the spec, the resulting apiVersion paramter is implicitly set by your SDK currently. 

```tsp
op valid(): NoContentResponse;
```

if you add apiVersion query parameter magically and someone generate an SDK but need that parameter they might not add it to their spec as the SDK "worked" and then removing/changing that behavior will break them.

Reply:
@pshao25 as far as I know we made an explicit choice to put emitters in @azure-tools namespace on npm to indicate these emitters are for azure sdk.  Usage outside of azure sdk is not explicitly supported although it might work in some cases.  This doesn't mean we will never support generating non azure sdk, but if we did I would imagine that emitter would go under the @typespec namespace.

@timotheeguerin do you think we should have a linter to just block the case where a customer doesn't define api-version  but is an azure service since this is a requirement for all azure services?  Or are you aware of a case of an azure service that doesn't have an api-version and passed api stewardship?